rag/llm_interface.py

```diff
@@ -0,0 +1,70 @@
+import os
+import openai
+import logging
+import yaml
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer, TextStreamer
+
+FILE_PATH = os.path.dirname(os.path.realpath(__file__))
+
+class LLMInterface:
+    def __init__(self) -> None:
+        pass
+
+    def query(self, system: str, user_message: str):
+        raise NotImplementedError()
+
+class HuggingFaceInterface(LLMInterface):
```

Comment:
why do we need `model_interface` and `llm_interface` both?

Reply:
Already nuke `model_interface` in main